### PR TITLE
Categorical generator and tests

### PR DESCRIPTION
**Pull request by lizlouise1335**
Generator file and test file for category subset generator

### CODE REVIEW

File: tests/test_categorical_generator.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+from synthetic_data.dataset_generators.categorical_generator import random_categorical
+
+
+class TestRandomsCategories(unittest.TestCase):
+    def setUp(self):
+        self.rng = Generator(PCG64(12345))
+        self.categories = ['X', 'Y', 'Z']
+
+    def test_default_return_validity(self):
+        result = random_categorical(self.rng, num_rows = 5)
```
Comment (JGSweets) [inline tests/test_categorical_generator.py:13]: let's make sure to test at least two variation of num_rows to ensure functionality

File: tests/test_categorical_generator.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import unittest
+import numpy as np
+from numpy.random import Generator, PCG64
+from synthetic_data.dataset_generators.categorical_generator import random_categorical
```
Comment (JGSweets) [inline tests/test_categorical_generator.py:4]: should be a space between first party (synthetic_data) imports and third party

File: tests/test_categorical_generator.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import unittest
+import numpy as np
```
Comment (JGSweets) [inline tests/test_categorical_generator.py:2]: need a space between built-in and third party libraries

File: synthetic_data/dataset_generators/categorical_generator.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import numpy as np
```
Comment (JGSweets) [inline synthetic_data/dataset_generators/categorical_generator.py:1]: let's move this into `distinct_generators` and move the previous.

Also, let's make sure the tests follow the same folder structure as well.

Comment (JGSweets) [reply synthetic_data/dataset_generators/categorical_generator.py:1]: The folder should be there now that #293 merged.

**Review by JGSweets — DISMISSED** [review]

File: synthetic_data/distinct_generators/categorical_generator.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    :type rng: numpy Generator
+    :param categories: a list of values that are allowed in a categorical or None
+    :type categories: string, None, optional
+    :param num_rows: the number of rows in np array generated
+    :type num_rows: int, optional
+
+    :return: np array of categories
+    """
+    if categories is None:
+        categories = ["A", "B", "C", "D", "E"]
```
Comment (taylorfturner) [inline synthetic_data/distinct_generators/categorical_generator.py:24]: thinking this should be the default for the parameter....

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: synthetic_data/distinct_generators/categorical_generator.py (hunk 5 above)
Comment (taylorfturner) [reply synthetic_data/distinct_generators/categorical_generator.py:24]: mutable... nevermind

**Review by taylorfturner — CHANGES_REQUESTED** [review]

**Review by taylorfturner — APPROVED** [review]

**Pull request merged**